ui/hooks/useAccountTotalFiatBalance.js

```diff
@@ -136,6 +144,34 @@ export const useAccountTotalFiatBalance = (
     ...tokenFiatBalances,
   ).toString(10);
 
+  // we need to append some values to tokensWithBalance for UI
+  // this code was ported from asset-list
+  tokensWithBalances.forEach((token) => {
+    // token.string is the balance displayed in the TokenList UI
+    token.string = roundToDecimalPlacesRemovingExtraZeroes(token.string, 5);
+  });
+
+  // to sort by fiat balance, we need to compute this at this level
+  tokensWithBalances.forEach((token) => {
+    const contractExchangeTokenKey = Object.keys(mergedRates).find((key) =>
+      isEqualCaseInsensitive(key, token.address),
+    );
+
+    const tokenExchangeRate =
+      contractExchangeTokenKey && mergedRates[contractExchangeTokenKey];
```

Comment:
I don't think this to be an O(n) search.  The asset controllers should return/store these addresses in checksum format.  So you should be able to look those up directly with something (untested) like:

```suggestion
    const tokenExchangeRate = mergedRates[toChecksumAddress(token.address)];
```

Reply:
Ty for this. Seems to be working fine, and much cleaner: https://github.com/MetaMask/metamask-extension/pull/27184/commits/42e5667727b9d62140052ac238cb09a26385063b